src/main.rs

```diff
@@ -4,62 +4,126 @@ use quicksilver::{
     Input, Result, Settings, Window,
 };
 
-use rand::distributions::{Distribution, Uniform};
+use rand::distributions::{Distribution, Uniform, WeightedIndex};
 use std::vec::Vec;
+use rand::prelude::ThreadRng;
+use lyon::{geom::LineSegment, math::point};
 
+#[derive(Debug)]
 struct Node {
     x: f32,
     y: f32,
 }
 
+const HEIGHT: f32 = 1024.0;
+const LENGTH: f32 = 768.0;
+
+impl Node {
+    pub fn gen(mut rng: ThreadRng) -> Self {
+        let x_dist = Uniform::from(0.0..HEIGHT);
+        let y_dist = Uniform::from(0.0..LENGTH);
+        Node {
+            x: x_dist.sample(&mut rng),
+            y: y_dist.sample(&mut rng),
+        }
+    }
+}
+
 struct World {
     nodes: Vec<Node>,
     connections: Vec<(u32, u32)>,
 }
 
-fn gen_world() -> World {
-    let node_count = 20;
+impl World {
+    pub fn gen() -> World {
+        let node_count = 20;
 
-    let mut nodes = Vec::new();
+        let mut rng = rand::thread_rng();
+        let nodes: Vec<Node> = (0..node_count)
+            .map(|_| Node::gen(rng))
+            .collect();
 
-    let mut rng = rand::thread_rng();
-    let x_dist = Uniform::from(0.0..1024.0);
-    let y_dist = Uniform::from(0.0..768.0);
-    for _ in 0..node_count {
-        let new_node = Node {
-            x: x_dist.sample(&mut rng),
-            y: y_dist.sample(&mut rng),
+        let connections_count = 40;
+        let mut connections: Vec<(u32, u32)>  = Vec::new();
+        for i in 0..connections_count {
+            let distrib = Uniform::from(0..nodes.len());
+            let node_index = distrib.sample(&mut rng);
+            let node = nodes.get(node_index).unwrap();
+            let weights: Vec<f32> = nodes.iter()
+                .map(|i| (euclidean_dist(node, i) / 1000.0).exp())
+                .collect();
+
+            let max_weights: f32 = weights.iter().fold(0.0f32, |val, next_val| next_val.max(val));
+            let normalized_weights: Vec<f32> = weights.into_iter()
+                .map(|elem| max_weights - elem)
+                .collect();
+
+            let sum_weights: f32 = normalized_weights.iter().sum();
+            let mut scaled_weights: Vec<f32> = normalized_weights.into_iter()
+                .map(|elem| elem / sum_weights)
+                .collect();
+
+            #[cfg(debug_assertions)]
+            if i == 0 {
+                println!("{:?}", &scaled_weights);
+            };
+
+
+            for (fst_connection, snd_connection) in connections.iter() {
+                let node1 = nodes.get(*fst_connection as usize).unwrap();
+                let node2 = nodes.get(*snd_connection as usize).unwrap();
+                let existed_line = LineSegment {
+                    from: point(node1.x, node1.y),
+                    to: point(node2.x, node2.y)
+                };
+                for (supposed_node_position, item) in scaled_weights.iter_mut().enumerate() {
```

Comment:
For discussion: IMHO there is an inverse: "for each connection check that for each node there is no intersection". More straightforward is "for each node check that there is no connection which we will intersect".

In the second case we can make some optimisation after set weight to `0.0`: just jump to next node, skipping other connections.
In the first case we will check node again and again despite it already has an intersection.